src/dodal/devices/dcm.py

```diff
@@ -0,0 +1,88 @@
+from bluesky.protocols import HasHints, Hints
+from ophyd_async.core import StandardReadable
+from ophyd_async.epics.motion import Motor
+from ophyd_async.epics.signal import epics_signal_r
+
+from dodal.beamlines.beamline_parameters import get_beamline_parameters
+
+
+class DCM(StandardReadable, HasHints):
+    """
+    A double crystal monochromator (DCM), used to select the energy of the beam.
+
+    perp describes the gap between the 2 DCM crystals which has to change as you alter
+    the angle to select the requested energy.
+
+    offset ensures that the beam exits the DCM at the same point, regardless of energy.
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        daq_configuration_path: str,
```

Comment:
@DominicOram and @rtuck99 how would you feel about moving the configuration file attributes for both the `DCM` and `Undulator` into `UndulatorDCM`? Since it's just a convenient place to store them for Hyperion and means that the devices are as simple as possible for the read-only use case

Reply:
That's fine by me

Reply:
Done, see what you think